doc: Update Kata Containers tutorial to ACRN v2.0

gvancuts opened the pull request:
Update the "Run Kata Containers on a Service VM" tutorial to use ACRN v2.0. It
also uses Ubuntu 20.04 as the baseline instead of Ubuntu 18.04. The Kata
Containers installation method is different as there are no pre-built binaries
for Ubuntu 20.04 yet, so using the 'kata-deploy' approach instead which is more
generic.

Tracked-On: #4944
Signed-off-by: Geoffroy Van Cutsem <geoffroy.vancutsem@intel.com>

gvancuts commented:
@ttzeng @vijaydhanraj PTAL

HuiHuangShi commented:
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

ttzeng commented on doc/tutorials/run_kata_containers.rst:
```diff
@@ -19,14 +19,15 @@ Prerequisites
+
 #. Kata Containers are only supported for ACRN hypervisors configured for
-   the SDC scenario.
+   the Industry scenario.
```
@gvancuts , per [[1]](https://github.com/projectacrn/acrn-hypervisor/blob/release_2.0/hypervisor/scenarios/sdc/vm_configurations.c#L51-L64) and [[2]](https://github.com/projectacrn/acrn-hypervisor/blob/release_2.0/hypervisor/scenarios/industry/vm_configurations.c#L123), it looks like both SDC and Industry scenarios are supported.
The rest of modifications look good, and just learned `kata-deploy` from you! Thank you (y)

gvancuts replied:
Very good point, I had missed that completely!

(note to self: I should also update the https://projectacrn.github.io/latest/introduction/index.html#best-known-configurations table)

gvancuts replied:
> (note to self: I should also update the https://projectacrn.github.io/latest/introduction/index.html#best-known-configurations table)

See #4950

gvancuts commented:
@dbkinder , there is a link to a presentation on Google Drive in this doc (see https://drive.google.com/file/d/1ZrqM5ouWUJA0FeIWhU_aitEJe8781rpe/view?usp=sharing). I couldn't find this presentation in Slideshare, did you put it there?

dbkinder commented:
I only did the 2020 presentations. I’ll add that one today and update the link.

-- david kinder

On Jun 16, 2020, at 3:14 AM, Geoffroy Van Cutsem <notifications@github.com> wrote:

﻿

@dbkinder<https://github.com/dbkinder> , there is a link to a presentation on Google Drive in this doc (see https://drive.google.com/file/d/1ZrqM5ouWUJA0FeIWhU_aitEJe8781rpe/view?usp=sharing). I couldn't find this presentation in Slideshare, did you put it there?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/projectacrn/acrn-hypervisor/pull/4949#issuecomment-644671872>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ACUNCQVQVKLFQYZB44N6SD3RW5AXDANCNFSM4N7M26IA>.

gvancuts commented:
> I only did the 2020 presentations. I’ll add that one today and update the link.

Thanks!

vijaydhanraj commented on doc/tutorials/run_kata_containers.rst:
```diff
@@ -19,14 +19,15 @@ Prerequisites
+   - ACRN v2.0 (branch: ``release_2.0``)
+   - Ubuntu 20.04
+
 #. Kata Containers are only supported for ACRN hypervisors configured for
```
Since ACRN supports two scenarios with Kata, "only" can be removed.

gvancuts replied:
OK

vijaydhanraj commented on doc/tutorials/run_kata_containers.rst:
```diff
@@ -58,41 +59,88 @@ Install Kata Containers
 
 Kata Containers provide a variety of installation methods, this guide uses
-:command:`kata-manager` to automate the Kata Containers installation procedure.
+`kata-deploy <https://github.com/kata-containers/packaging/tree/master/kata-deploy>`_
```
Haven't tried kata-deploy, but what is the motivation to switch from kata-manager to kata-deploy?

gvancuts replied:
There are no builds for Ubuntu 20.04 (which I am using)... so `kata-manager` fails. See
* https://github.com/kata-containers/ci/issues/277
* https://github.com/kata-containers/tests/issues/1691

That's when I decided to use `kata-deploy`, which is rather straightforward to use and can work with both Ubuntu 18.04 and 20.04 (and I guess pretty much any other Linux distro too).

vijaydhanraj commented on doc/tutorials/run_kata_containers.rst:
```diff
@@ -58,41 +59,88 @@ Install Kata Containers
       {
         "storage-driver": "devicemapper",
-        "default-runtime": "kata-runtime",
+        "default-runtime": "kata-acrn",
```
why rename the runtime? If kata generates runtime as kata-runtime it is better to use it that way.

HuiHuangShi commented:
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

dbkinder commented:
@gvancuts I updated/force pushed to your PR with the slideshare.net presentation link, and fixed a misspelling while I was in there :)

gvancuts commented:
> @gvancuts I updated/force pushed to your PR with the slideshare.net presentation link, and fixed a misspelling while I was in there :)

Thanks!! 👍

HuiHuangShi commented:
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>
